src/gen_smtp_server_session.erl

```diff
@@ -66,6 +66,7 @@
 		ranch_version :: lt16 | gte16,
 		envelope = undefined :: 'undefined' | #envelope{},
 		extensions = [] :: [{string(), string()}],
+		maxsize = ?MAXIMUMSIZE,
```

Comment:
would be nice to have a typespec. `:: non_neg_integer()` should be good.

Reply:
Yes, I changed it to `pos_integer() | 'infinity'`

Note that the callback server code has to return `{"SIZE", "0"}` and know to map `infinity` to `0`.  I guess that is ok, as you have to be prepared to handle unlimited size anyway.